src/SyntaxHighlighter/Cpp.cc

```diff
@@ -0,0 +1,103 @@
+const char* const kKeywordRegex =
+    "\\b(?:__attribute__|_Alignas|_Alignof|_Atomic|_Bool|_Complex|_Generic|_Imaginary|_Noreturn|_"
+    "Static_assert|_Thread_local|asm|typeof|inline|auto|break|case|char|const|continue|default|do|"
+    "double|else|enum|extern|float|for|goto|if|int|long|register|return|short|signed|sizeof|static|"
+    "struct|switch|typedef|union|unsigned|void|volatile|while|)\\b";
+const char* const kCommentRegex =
+    "\\/\\/(?:[^\\r\\n\\\\]|\\\\(?:\\r\\n?|\\n|(?![\\r\\n])))*|\\/\\*[\\s\\S]*?\\*\\/";
+const char* const kOpenCommentRegex = "\\/\\*([^\\*]|[\\*]+[^\\/])*?$";  // Don't work with **/
+const char* const kEndCommentRegex = "[\\s\\S]*\\*\\/";
+const char* const kFunctionRegex = "[a-z_]\\w*(?=\\s*\\()";
+const char* const kNoEndCommentRegex = "([^\\*]|\\*+[^\\/\\*])*$";
+const char* const kMacroRegex = "(^\\s*)#\\s*[a-z_]\\S*";
+const char* const kIncludeFileRegex = "^(#\\s*include\\s*)<[^>]+>";
+const char* const kStringRegex = "\"([^\\\\\"]|\\\\.)*\"|\'[^\']*\'";
+const char* const kOpenStringRegex = "\"([^\\\\\"]|\\\\.)*\\\\$";
+const char* const kEndStringRegex = "([^\\\\\"]|\\\\.)*\"";
+const char* const kNoEndStringRegex = "([^\\\\\"]|\\\\.)*\\\\$";
+const char* const kUppercaseRegex = "[A-Z][\\w\\*\\&]*";
+}  // namespace C
+
+Cpp::Cpp() : QSyntaxHighlighter{static_cast<QObject*>(nullptr)} {
+  using PatternOption = QRegularExpression::PatternOption;
+  number_regex_ = QRegularExpression{C::kNumberRegex, PatternOption::CaseInsensitiveOption};
+  constant_regex_ = QRegularExpression{C::kConstantRegex};
+  keyword_regex_ = QRegularExpression{C::kKeywordRegex};
+  comment_regex_ = QRegularExpression{C::kCommentRegex};
+  open_comment_regex_ = QRegularExpression{C::kOpenCommentRegex};
+  end_comment_regex_ = QRegularExpression{C::kEndCommentRegex};
+  no_end_comment_regex_ = QRegularExpression{C::kNoEndCommentRegex};
+  function_regex_ = QRegularExpression{C::kFunctionRegex, PatternOption::CaseInsensitiveOption};
+  macro_regex_ = QRegularExpression{C::kMacroRegex};
+  include_file_regex_ = QRegularExpression{C::kIncludeFileRegex};
+  string_regex_ = QRegularExpression{C::kStringRegex};
+  open_string_regex_ = QRegularExpression{C::kOpenStringRegex};
+  end_string_regex_ = QRegularExpression{C::kEndStringRegex};
+  no_end_string_regex_ = QRegularExpression{C::kNoEndStringRegex};
+  uppercase_regex_ = QRegularExpression{C::kUppercaseRegex};
+}
+
+void Cpp::highlightBlock(const QString& code) {
+  const auto apply = [&code, this](QRegularExpression* expression, const QColor& color,
+                                   int new_state = CppHighlighterState::kInitialState) {
+    QTextCharFormat format{};
+    format.setForeground(color);
+
+    for (auto it = expression->globalMatch(code); it.hasNext();) {
+      const auto match = it.next();
+      setFormat(match.capturedStart(), match.capturedLength(), format);
+      setCurrentBlockState(new_state);
+    }
+  };
+
+  apply(&number_regex_, C::kBlue);
```

Comment:
I guess long term, we want to have central place for all colors. For now, I think it is fine as it is.

Reply:
Yes, completely agree. Is also one of the things in my roadmap that I want to tackle.